File: model/request_ad_equipment_client.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type RequestAdEquipmentClient struct {
	ID                  int               `json:"id"`
	CreatedAt           time.Time         `json:"created_at"`
	UpdatedAt           time.Time         `json:"updated_at"`
	DeletedAt           gorm.DeletedAt    `json:"deleted_at"             swaggertype:"string"`
	AdEquipmentClientID int               `json:"ad_equipment_client_id"`
	AdEquipmentClient   AdEquipmentClient `json:"ad_equipment_client"`
	UserID              int               `json:"user_id"`
	User                User              `json:"user"`
	ExecutorID          *int              `json:"executor_id"`
	Executor            User              `json:"executor"               gorm:"foreignKey:ID;references:ExecutorID"`
	Status              string            `json:"status"`
	Description         string            `json:"description"`
}

type FilterRequestAdEquipmentClient struct {
	AdEquipmentClientDetail         *bool
	AdEquipmentClientDocumentDetail *bool
	UserDetail                      *bool
	ExecutorDetail                  *bool
	Unscoped                        *bool
	Limit                           *int
	Offset                          *int
	Status                          *string
	Description                     *string
	IDs                             []int
	AdEquipmentClientIDs            []int
	UserIDs                         []int
	ExecutorIDs                     []int
}

File: mobile/lib/src/features/requests/data/models/equipment_request/equipment_client_request_model/request_ad_equipment_client.dart
import 'package:eqshare_mobile/src/core/data/models/user_model/user.dart';
import 'package:eqshare_mobile/src/features/ad/data/models/ad_equipment_models/ad_equipment_client_model/ad_equipment_client.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'request_ad_equipment_client.freezed.dart';
part 'request_ad_equipment_client.g.dart';

@freezed
class RequestAdEquipmentClient with _$RequestAdEquipmentClient {
  factory RequestAdEquipmentClient({
    int? id,
    @JsonKey(name: 'created_at') DateTime? createdAt,
    @JsonKey(name: 'updated_at') DateTime? updatedAt,
    @JsonKey(name: 'deleted_at') dynamic? deletedAt,
    @JsonKey(name: 'ad_equipment_client_id') int? adEquipmentClientId,
    @JsonKey(name: 'ad_equipment_client') AdEquipmentClient? adEquipmentClient,
    @JsonKey(name: 'user_id') int? userId,
    @JsonKey(name: 'executor_id') int? executorId,
    @JsonKey(name: 'executor') User? executor,
    @JsonKey(name: 'user') User? user,
    String? status,
    String? description,
  }) = _RequestAdEquipmentClient;

  factory RequestAdEquipmentClient.fromJson(Map<String, dynamic> json) =>
      _$RequestAdEquipmentClientFromJson(json);
}
